src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ManifestMetadataTableNode.cs

```diff
@@ -62,37 +62,53 @@ public class ManifestMetadataTableNode : HeaderTableNode
         /// </summary>
         private bool _emissionCompleted;
 
-        /// <summary>
-        /// Name of the input assembly.
-        /// </summary>
-        private string _inputModuleName;
-
         /// <summary>
         /// Node factory for the compilation
         /// </summary>
         private readonly NodeFactory _nodeFactory;
 
-        public ManifestMetadataTableNode(EcmaModule inputModule, NodeFactory nodeFactory)
-            : base(inputModule.Context.Target)
+        public ManifestMetadataTableNode(NodeFactory nodeFactory)
+            : base(nodeFactory.Target)
         {
             _assemblyRefToModuleIdMap = new Dictionary<string, int>();
             _moduleIdToAssemblyNameMap = new Dictionary<int, AssemblyName>();
             _signatureEmitters = new List<ISignatureEmitter>();
             _nodeFactory = nodeFactory;
 
-            _inputModuleName = inputModule.Assembly.GetName().Name;
-
-            _assemblyRefCount = inputModule.MetadataReader.GetTableRowCount(TableIndex.AssemblyRef);
-            for (int assemblyRefIndex = 1; assemblyRefIndex <= _assemblyRefCount; assemblyRefIndex++)
+            if (!_nodeFactory.CompilationModuleGroup.IsCompositeBuildMode)
             {
-                AssemblyReferenceHandle assemblyRefHandle = MetadataTokens.AssemblyReferenceHandle(assemblyRefIndex);
-                AssemblyReference assemblyRef = inputModule.MetadataReader.GetAssemblyReference(assemblyRefHandle);
-                string assemblyName = inputModule.MetadataReader.GetString(assemblyRef.Name);
-                _assemblyRefToModuleIdMap[assemblyName] = assemblyRefIndex;
+                if (_nodeFactory.CompilationModuleGroup.CompilationModuleSet.Skip(1).Any())
+                {
+                    throw new InternalCompilerErrorException("Multiple input files specified without the composite flag");
```

Comment:
Nit: This feels like something we should catch when setting up the `CompilationModuleGroup` so we can emit a localized error to the console and maybe just assert here.

Reply:
Good point, fixed.